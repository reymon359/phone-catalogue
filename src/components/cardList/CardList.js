import React from 'react';
import Card from '../card/Card';

const CardList = ({ phones }) => {
    return (
        <div>
            {phones.map((phone, i) => (
                <Card phone={phone} key={i}/>
            ))}
        </div>
    );
};

export default CardList;