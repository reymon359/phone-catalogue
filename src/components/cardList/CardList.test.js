import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('renders CardList component', () => {
    const mockPhones = [
        {
            id: 1,
            name: 'Nokia',
            price: '2',
            imageFileName: 'http://localhost:3002/images/Honor_10.png'
        }
    ];
    expect(shallow(<CardList phones={mockPhones} />)).toMatchSnapshot();
});
