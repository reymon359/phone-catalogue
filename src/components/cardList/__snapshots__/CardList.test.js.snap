// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    phones={
      Array [
        Object {
          "id": 1,
          "imageFileName": "http://localhost:3002/images/Honor_10.png",
          "name": "Nokia",
          "price": "2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          phone={
            Object {
              "id": 1,
              "imageFileName": "http://localhost:3002/images/Honor_10.png",
              "name": "Nokia",
              "price": "2",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "phone": Object {
            "id": 1,
            "imageFileName": "http://localhost:3002/images/Honor_10.png",
            "name": "Nokia",
            "price": "2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            phone={
              Object {
                "id": 1,
                "imageFileName": "http://localhost:3002/images/Honor_10.png",
                "name": "Nokia",
                "price": "2",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "phone": Object {
              "id": 1,
              "imageFileName": "http://localhost:3002/images/Honor_10.png",
              "name": "Nokia",
              "price": "2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
